life_expectancy/cleaning.py

```diff
@@ -1,133 +1,110 @@
+                                  var_name = 'year',
+                                  value_name = 'value')
 
+    #Set and check data types
+    dataframe_correct_dtypes = unpivoted_dataframe
     #Cast year column as integer
-    data_types_dataframe['year'] = data_types_dataframe['year'].astype('int')
-
-    #Cast value column as float (deal with null values)
-    data_types_dataframe['value'] = data_types_dataframe['value'].str.split(' ').str[0]
-
-    data_types_dataframe['value'] = data_types_dataframe['value'].replace({':': 0, ': ': 0})
-    data_types_dataframe['value'] = data_types_dataframe['value'].astype('float')
-    data_types_dataframe = data_types_dataframe[data_types_dataframe['value'] != 0]
-
-    data_types_dataframe.columns = ['unit', 'sex', 'age', 'region', 'year', 'value']
-
-    return data_types_dataframe
-
-def filter_by_region(
-    all_regions_dataframe: DataFrame,
-    region: str
-):
-    """
-    This function filters the provided dataframe by the specified region
-
-    :param all_regions_dataframe -> dataframe with every region's data
-    :param region -> region to filter the data by
-
-    return region_filtered_dataframe -> dataframe filtered by region
-    """
-
-    region_filtered_dataframe = all_regions_dataframe[all_regions_dataframe['region'] == region]
-
-    return region_filtered_dataframe
+    dataframe_correct_dtypes['year'] = dataframe_correct_dtypes['year'].astype('int')
+    #Cast value column as float
+    dataframe_correct_dtypes['value'] = dataframe_correct_dtypes['value'].str.split(' ').str[0]
+    #Deal with null values
+    dataframe_correct_dtypes['value'] = dataframe_correct_dtypes['value'].replace({':': -1,
+                                                                                   ': ': -1})
+    dataframe_correct_dtypes['value'] = dataframe_correct_dtypes['value'].astype('float')
+    dataframe_correct_dtypes = dataframe_correct_dtypes[dataframe_correct_dtypes['value'] != -1]
+    #Rename columns
+    dataframe_correct_dtypes.columns = ['unit', 'sex', 'age', 'region', 'year', 'value']
+
+    #Filter by region
+    clean_dataframe = dataframe_correct_dtypes[dataframe_correct_dtypes['region'] == country]
+
+    return clean_dataframe
 
 def save_data(
-    filtered_dataframe: DataFrame,
-    safe_path: str
+    clean_dataframe: DataFrame,
 ):
     """
     This function saves the final dataframe in the desired path
+    param: clean_dataframe -> Cleaned pandas dataframe
     """
-    os_path_pardir = os.path.dirname(__file__)
 
-    filtered_dataframe.to_csv(os_path_pardir + safe_path, index = False)
-
-def clean_data(country: str = 'PT'):
-    """
-    This function performs the full data cleaning process
-    """
-    raw_data_file_path = '/data/eu_life_expectancy_raw.tsv'
-    final_data_file_path = '/data/pt_life_expectancy.csv'
-    raw_dataframe = load_raw_data(raw_data_file_path)
+    #Get parent directory
+    parent_directory = os.path.dirname(__file__)
```

Comment:
parent directory shall be defined for both this function and the one that reads the data. Defining it twice, inside each function makes it a code smell